### PR TITLE
Properly handle Aliro credential validation in door lock server

### PR DESCRIPTION
**Pull request by krypton36**
The door locker server currently does not have a way to validate Aliro credentials and fails when attempting to add a credential with the correct parameters. This PR adds proper validation of the Aliro credential to the door lock server.

### CODE REVIEW

File: src/app/clusters/door-lock-server/door-lock-server.cpp [hunk 1]
```diff
@@ -44,8 +44,9 @@ using namespace chip::app::Clusters::DoorLock;
 using namespace chip::app::Clusters::DoorLock::Attributes;
 using chip::Protocols::InteractionModel::Status;
 
-static constexpr uint8_t DOOR_LOCK_SCHEDULE_MAX_HOUR   = 23;
-static constexpr uint8_t DOOR_LOCK_SCHEDULE_MAX_MINUTE = 59;
+static constexpr uint8_t DOOR_LOCK_SCHEDULE_MAX_HOUR        = 23;
+static constexpr uint8_t DOOR_LOCK_SCHEDULE_MAX_MINUTE      = 59;
+static constexpr uint8_t DOOR_LOCK_ALIRO_CREDENTIAL_MIN_MAX = 65;
```
Comment (bzbarsky-apple) [inline src/app/clusters/door-lock-server/door-lock-server.cpp:49]:
```suggestion
static constexpr uint8_t DOOR_LOCK_ALIRO_CREDENTIAL_SIZE = 65;
```
I guess file style is SHOUTY_SNAKE_CASE, ok.

**Review by bzbarsky-apple — APPROVED** [review]

File: src/app/clusters/door-lock-server/door-lock-server.cpp (hunk 1 above)
Comment (krypton36) [reply src/app/clusters/door-lock-server/door-lock-server.cpp:49]: 🤣

File: src/app/clusters/door-lock-server/door-lock-server.h [hunk 2]
```diff
@@ -184,6 +184,8 @@ class DoorLockServer : public chip::app::AttributeAccessInterface
 
     inline bool SupportsFace(chip::EndpointId endpointId) { return GetFeatures(endpointId).Has(Feature::kFaceCredentials); }
 
+    inline bool SupportsAliro(chip::EndpointId endpointId) { return GetFeatures(endpointId).Has(Feature::kAliroProvisioning); }
```
Comment (nivi-apple) [inline src/app/clusters/door-lock-server/door-lock-server.h:187]: There is already a SupportsAliroProvisioning API. Cna you use that?

**Review by nivi-apple — COMMENTED** [review]
Looks good. Just use the existing API for checking if aliro is supported

File: src/app/clusters/door-lock-server/door-lock-server.h (hunk 2 above)
Comment (krypton36) [reply src/app/clusters/door-lock-server/door-lock-server.h]: Updated. Thank you

**Review by nivi-apple — APPROVED** [review]

**Pull request merged**